tt_metal/host_api.hpp

```diff
@@ -259,6 +259,17 @@ uint32_t CreateSemaphore(
 */
 std::shared_ptr<Buffer> CreateBuffer(const InterleavedBufferConfig &config);
 
+/**
+*  Creates a pre-allocated interleaved DRAM or L1 buffer on device
+*
+*  Return value: std::shared_ptr<Buffer>
+*
+*  | Argument        | Description                             | Type                     | Valid Range | Required |
+*  |-----------------|---------------------------------------- |--------------------------|-------------|----------|
+*  | config          | config for buffer                       | InterleavedBufferConfig  |             | Yes      |
```

Comment:
should probably describe address field on these new APIs

Reply:
Added the description